broker/utils_test.go

```diff
@@ -13,54 +13,50 @@ import (
 var _ = Describe("Broker utility functions", func() {
 
 	Describe("findDeployment", func() {
-		var (
-			fakeComposeClient *fakes.FakeComposeClient
-		)
-
-		BeforeEach(func() {
-			fakeComposeClient = fakes.New()
-			fakeComposeClient.Deployments = []composeapi.Deployment{
-				{ID: "1234", Name: "one"},
-				{ID: "2345", Name: "two"},
-			}
-		})
-
 		It("returns the deployment if present", func() {
-			d, err := findDeployment(fakeComposeClient, "two")
-			Expect(err).NotTo(HaveOccurred())
-			Expect(d.ID).To(Equal("2345"))
-			Expect(d.Name).To(Equal("two"))
+			fakeComposeClient := &fakes.FakeClient{}
+			deployment := &composeapi.Deployment{ID: "2345", Name: "two"}
+			fakeComposeClient.GetDeploymentByNameReturns(deployment, []error{})
+
+			Expect(findDeployment(fakeComposeClient, "two")).To(Equal(deployment))
 		})
 
 		It("returns a errDeploymentNotFound if the deployment doesn't exist", func() {
+			fakeComposeClient := &fakes.FakeClient{}
+			fakeComposeClient.GetDeploymentByNameReturns(nil, []error{errors.New("deployment not found")})
+
 			_, err := findDeployment(fakeComposeClient, "non-existent")
+
 			Expect(err).To(Equal(errDeploymentNotFound))
 		})
 
 		It("returns all other errors", func() {
-			fakeComposeClient.GlobalError = errors.New("computer says no")
+			fakeComposeClient := &fakes.FakeClient{}
+			fakeComposeClient.GetDeploymentByNameReturns(nil, []error{errors.New("computer says no")})
+
 			_, err := findDeployment(fakeComposeClient, "one")
-			Expect(err).To(Equal(fakeComposeClient.GlobalError))
+
+			Expect(err).To(MatchError("computer says no"))
 		})
 	})
 
 	Describe("makeOperationData", func() {
 		It("can make operation data JSON", func() {
-			operationData, err := makeOperationData("expected_type", "123")
+			operationData, err := makeOperationData("expected_type", "123", []string{"1.1.1.1"})
```

Comment:
Does it seem reasonable to use IPs in subnets defined in [RFC5737](https://tools.ietf.org/html/rfc5737) instead? Or possibly a bunch of things on the loopback subnet? `127.0.0.0/8`? This possibly applies to other hard-coded IPs in this PR.

Reply:
(The PR looks good otherwise and I'm not sold on this if you don't feel like it's worth the effort)

Reply:
I don't feel strongly about altering it for the unit tests tbh since it's really just testing that the string comes out the other side.

but I wasn't aware of those blocks... I'll def use them in the future for tests... good tip.

Reply:
Cool. I'll merge.